pkg/controllers/binding/common_test.go

```diff
@@ -379,3 +379,154 @@ func Test_shouldSuspendDispatching(t *testing.T) {
 		})
 	}
 }
+
+func Test_needReviseReplicas(t *testing.T) {
+	tests := []struct {
+		name      string
+		replicas  int32
+		placement *policyv1alpha1.Placement
+		want      bool
+	}{
+		{
+			name:     "replicas is zero",
```

Comment:
@yashpandey06 You can add more test cases like with very large replica count (math.MaxInt32) and with negative replica count (you can skip it as replicas cannot be negative in a real-world Kubernetes context and it makes no sense).